### PR TITLE
REFACTOR: Remove Duplicated Logit

### PR DESCRIPTION
**Pull request by marload**
I removed duplication and improved readability by refactoring the code.

Thank You!

### CODE REVIEW

File: tensorflow_addons/layers/adaptive_pooling.py [hunk 1]
```diff
@@ -53,16 +53,13 @@ def __init__(
         super().__init__(**kwargs)
 
     def call(self, inputs, *args):
-        bins = self.output_size[0]
```
Comment (bhack) [inline tensorflow_addons/layers/adaptive_pooling.py:56]: Here bins probably has more semantic info.

Comment (marload) [reply tensorflow_addons/layers/adaptive_pooling.py:56]: I agree.
Thank You!

**Comment by Susmit-A** [comment]
I believe it makes the logic sligtly less obvious to readers. Also, pooling in higher dimensions can't be simplified as neatly as in case of 1D pooling, so this would break the trend of code applied in all layers.

**Comment by bhack** [comment]
@WindQAQ are we deprecating data_format also in case like this one as in https://github.com/tensorflow/addons/issues/1825?

**Comment by WindQAQ** [comment]
> @WindQAQ are we deprecating `data_format` also in case like this one as in #1825?

No, that issue is specific to `tfa.image`.

> I believe it makes the logic sligtly less obvious to readers. Also, pooling in higher dimensions can't be simplified as neatly as in case of 1D pooling, so this would break the trend of code applied in all layers.

I agree with @Susmit-A. IMO, splitting cases *clearly* is much more important than duplicated codes elimination.

**Comment by bhack** [comment]
More in general, what I think is that when we have some component with a "family" already in Tensorflow we could check TF style before submit small style refactoring PR. Also if currently it is not a mandatory check in our contributing guidelines or PR template /cc @seanpmorgan. 
E.g. in this case https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/pooling.py#L83

**Comment by marload** [comment]
I will close this PR!

**Pull request closed without merging**